edx_event_bus_kafka/internal/consumer.py

```diff
@@ -282,11 +287,88 @@ def record_event_consuming_error(self, run_context, error, maybe_event):
             # and will only read the exception from stack context.
             raise Exception(error)
         except BaseException:
+            self._add_message_monitoring(run_context=run_context, message=maybe_event, error=error)
             record_exception()
             logger.exception(
                 f"Error consuming event from Kafka: {error!r} in context {context_msg} -- {event_msg}"
             )
 
+    def _add_message_monitoring(self, run_context, message, error=None):
+        """
+        Record additional details for monitoring.
+
+        Arguments:
+            run_context: Dictionary of contextual information: full_topic, consumer_group,
+              and expected_signal.
+            message: None if event could not be fetched or decoded, or a Message if one
+              was successfully deserialized but could not be processed for some reason
+            error: (Optional) An exception instance, or None if no error.
+        """
+        try:
+            kafka_message, kafka_error = self._get_kafka_message_and_error(message=message, error=error)
+
+            # .. custom_attribute_name: kafka_topic
+            # .. custom_attribute_description: The full topic of the message or error.
+            set_custom_attribute('kafka_topic', run_context['full_topic'])
+            # .. custom_attribute_name: kafka_has_message
+            # .. custom_attribute_description: Boolean describing whether there was a message involved. Some errors
+            #   have no message.
+            set_custom_attribute('kafka_has_message', bool(kafka_message))
+
+            if kafka_message:
+                headers = kafka_message.headers() or []  # treat None as []
+                # header is list of tuples, so handle case with duplicate headers for same key
+                message_ids = [value.decode("utf-8") for key, value in headers if key == EVENT_ID_HEADER]
+                if message_ids and len(message_ids) > 0:
```

Comment:
Nit: do we need a null check for a list comprehension?

Reply:
Done.